Sources/NIOSSL/ObjectIdentifier.swift

```diff
@@ -0,0 +1,110 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_implementationOnly import CNIOBoringSSL
+@_implementationOnly import CNIOBoringSSLShims
+
+/// Object Identifier (OID)
+public struct NIOSSLObjectIdentifier: Hashable {
+    fileprivate final class Storage {
+        
+        /// All operations accessing `reference` need to be implemented while guaranteeing that the enclosing `self` has a reference count of at least one.
+        /// Otherwise `reference` could already be freed. This would result in undefined behaviour as we access a dangling pointer.
+        /// One way we can guarantee this is by implementation all operations as methods/computed properties on `Storage`.
+        /// Swift guarantees that `self` is not deallocated during the execution of a method/computed property on `self`.
+        private let reference: OpaquePointer
+        
+        fileprivate init?(_ string: String) {
+            let result = string.withCString { string in
+                CNIOBoringSSL_OBJ_txt2obj(string, 0)
```

Comment:
Are we deliberately setting this to 0? Should we only tolerate the dotted sequence of numbers?

Reply:
I think we only need to support the dotted sequence of numbers. Changed the setting to `1` and copied the documentation from OpenSSL inline.